### DISCUSSION TITLE
Disabling window snap-grab?

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#261] [Q&A] [answered]
Is there a way to disable the grab mode where having your hand close to the window will show the line to the title center and grab the window if you pinch your fingers? I find it interferes quite often with interacting with buttons near the window top.

**Comment by maluoi** [comment, marked as answer]
There's a [`UI.EnableFarInteract`](https://stereokit.net/Pages/Reference/UI/EnableFarInteract.html) property that'll work for this! This can work globally by just setting it once, as well as selectively, buy turning it on and off around relevant elements.

```csharp
UI.EnableFarInteract = false;
UI.WindowBegin();
UI.WindowEnd();
UI.EnableFarInteract = true;
```

That being said, far grab should disable when the hand is close to the window, or the window is within easy reach. It's possible there could be something buggy or excitable going on there that I should look into and refine! Is this a really big or wide window by any chance?

**Reply by paulmelis** [reply]
Well, the windows contain quite a few buttons, and so need to be somewhat wide (40-60 cm). 

At what distance should the far grab disable? The behaviour I see in the HL is that the far grab line appears when my hand is roughly 50cm from the window, and it then stays active all the way to roughly 2cm distance from the window surface, which is really close and only just before starting to e.g. press a button.

**Reply by paulmelis** [reply]
Okay, disabling the far interact with the flag is much better, thanks! Now if I only could find the same flag for the Windows OS itself, as it keeps using far interaction even when my hand is close to application windows ;-)
